packages/core/src/suggestion/SuggestionTask.ts

```diff
@@ -0,0 +1,34 @@
+import { performance } from 'perf_hooks';
+
+import { Suggestion } from '../models';
+import { SuggestionInput } from '../types';
+
+export abstract class SuggestionTask {
+  name: string = '';
+  elapsedTime: number = 0;
+
+  abstract task(suggestionInput: SuggestionInput): Promise<Suggestion>;
+
+  // Handles running the task and getting the elapsed time of the task
+  async run(suggestionInput: SuggestionInput): Promise<Suggestion> {
+    if (typeof this?.task !== 'function') {
+      throw new Error(
+        'Please implement task on your subclass of SuggestionTask'
+      );
+    }
```

Comment:
_technically_ this isnt needed since people are extending from this abstract class and are required to implement abstract properties, we can clean this up later